src/Streamly/Prelude.hs

```diff
@@ -120,6 +123,20 @@ unfoldrM step = fromStream . go
 each :: (Streaming t, Foldable f) => f a -> t m a
 each = Prelude.foldr cons nil
 
+-- | Iterate a pure function from a seed value, streaming the results forever
+iterate :: Streaming t => (a -> a) -> a -> t m a
```

Comment:
BTW, in general, for any pure style functions we can use the list functions and convert that to a stream e.g. `each $ iterate f x` should be equivalent to `iterate f x`. I am not sure if the native iterate is significantly more efficient compared to that?